src/utils.rs

```diff
@@ -45,9 +46,9 @@ fn dummy_callback(_: LogVerbosity, _: String) {}
 /// with nice Rust inputs.
 unsafe extern "C" fn log_callback(importance: wlr_log_importance,
                                   fmt: *const c_char,
-                                  _va_list: *mut __va_list_tag) {
-    RUST_LOGGING_FN(importance,
-                    c_to_rust_string(fmt).unwrap_or_else(|| "".into()))
+                                  va_list: *mut __va_list_tag) {
+    let message = vsprintf(fmt, va_list).unwrap_or_else(|_| "".to_string());
```

Comment:
Can we actually return the `fmt` string on failure? That will be clearer than an empty line in the log.